perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java

```diff
@@ -852,29 +852,54 @@ private void setFromAndReplyTo(MimeMessage message, String fromName, String from
 
 	/**
 	 * Get proper values "TO" for mail message based on VO or GROUP attribute "toEmail"
-	 *
+	 * <p>
 	 * If group attribute not set and is group application, get vo attribute as backup.
 	 * If no attribute set, BACKUP_FROM address will be used.
 	 *
 	 * @param app application to decide if it's VO or Group application
-	 *
 	 * @return list of mail addresses to send mail to
 	 */
-	private List<String> getToMailAddresses(Application app){
+	private List<String> getToMailAddresses(Application app) {
 		List<String> result = new ArrayList<>();
 
 		// get proper value from attribute
 		try {
-			Attribute attrToEmail = getMailFromVoAndGroupAttrs(app, URN_VO_TO_EMAIL, URN_GROUP_TO_EMAIL);
+			Attribute attrSenderEmail;
+
+			if (app.getGroup() == null) {
+				// it is a VO application
+				return getMailsFromVo(app, URN_VO_TO_EMAIL);
+			}
+
+			attrSenderEmail = attrManager.getAttribute(registrarSession, app.getGroup(), URN_GROUP_TO_EMAIL);
+
+			if (attrSenderEmail == null || attrSenderEmail.getValue() == null) {
+				// not specified toEmail group attribute
+
+				// try to use all group admins with specified preferred emails
+				List<User> admins = groupsManager.getAdmins(registrarSession, app.getGroup(), true);
```

Comment:
This is an old method that was used to get admins. We have a new general method in `AuthzResolverBlImpl.getRichAdmins(...)`. You could use the method here like this:
```suggestion
				List<RichUser> admins = AuthzResolverBlImpl.getRichAdmins(registrarSession, app.getGroup(), Collections.emptyList(), "GROUPADMIN", false, false);
```
This method takes a lot of arguments. If you want, you can add an overloaded version of this method:
```Java
public static List<RichUser> getRichAdmins(PerunSession sess, PerunBean complementaryObject, String role) throws RoleCannotBeManagedException {
	return getRichAdmins(sess, complementaryObject, Collections.emptyList(), role, false, false);
}
```
This would simplify the call here and also the one you will need to get Vo admins.